### ISSUE TITLE
tmPenalties is an array when blank (should be object)

### ISSUE THREAD

**Issue by yevheniilavrenchuk** [crowdin/crowdin-api-client-dotnet#247]
We are using the latest API released - 2.23.1
![image](https://github.com/crowdin/crowdin-api-client-dotnet/assets/137056044/ffd48657-8c24-4918-a788-be442b11ad64)

When Penalties are not filled in the Project settings, then in the API response it is an array:

![image](https://github.com/crowdin/crowdin-api-client-dotnet/assets/137056044/f135a43a-a20f-4356-ae40-a2cd3f676d83)

![image](https://github.com/crowdin/crowdin-api-client-dotnet/assets/137056044/2bbdb3d1-552f-48fe-93b6-4b753f5ece88)

But in API it is object:
![image](https://github.com/crowdin/crowdin-api-client-dotnet/assets/137056044/9cee25fc-6b80-4d10-9f8a-3af3298ccd0d)

When the request is made by the application, the error is thrown:
```
System.AggregateException: 'One or more errors occurred. (Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'Crowdin.Api.ProjectsGroups.TmPenalties' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
```

**Comment by andrii-bodnar** [comment]
This is a common thing for the API response when it returns an empty array (on the API side there is no difference between an empty object and an empty array).

We already have `EmptyArrayAsObjectConverter` in this client. This should probably be used in the `TmPenalties` case as well.

**Comment by innomaxx** [comment]
> This is a common thing for the API response when it returns an empty array (on the API side there is no difference between an empty object and an empty array).
> 
> 
> 
> We already have `EmptyArrayAsObjectConverter` in this client. This should probably be used in the `TmPenalties` case as well.
> 
> 
> 
> @innomaxx what do you think?

Yes, we can try fix it via custom converter, like in other similar cases, but it's a workaround rather than a long-term solution. I still recommend fix it API-widely on the Crowdin Backend side (if possible).

**Comment by andrii-bodnar** [comment]
Unfortunately, there is no quick general fix for the API. Each case should be fixed individually. It will take a lot of time to fix it (creating a task, scheduling, reaching the sprint, etc), so it would be better to fix it on the API client side using the custom converter.

**Comment by innomaxx** [comment]
Ok I'll handle it until asap

**Comment by andrii-bodnar** [comment]
@yevheniilavrenchuk available in [2.24.0](https://github.com/crowdin/crowdin-api-client-dotnet/releases/tag/2.24.0)

**Comment by yevheniilavrenchuk** [comment]
> @yevheniilavrenchuk available in [2.24.0](https://github.com/crowdin/crowdin-api-client-dotnet/releases/tag/2.24.0)

Thanks, it works like a charm!
